datastore-v1-proto-client/src/main/java/com/google/datastore/v1/client/EndToEndChecksumHandler.java

```diff
@@ -74,4 +70,12 @@ static boolean hasChecksumHeader(HttpResponse response) {
   static String getChecksumHeader(HttpResponse response) {
     return response.getHeaders().getFirstHeaderStringValue(HTTP_RESPONSE_CHECKSUM_HEADER);
   }
+
+  void update(byte[] bytes, int off, int len) {
```

Comment:
should these be `protected`?

Reply:
These 2 methods - update(..), hash() - are used by ChecksumEnforcingInputStream.
thats why I had to make them package scoped.